### DISCUSSION TITLE
LocationManager as @Dependency

### DISCUSSION THREAD

**Discussion by knight6700** [pointfreeco/composable-core-location#31] [Q&A]
how i can add LocationManager as @Dependency with TCA new Pattern

**Comment by mbrandonw** [comment]
Hi @knight6700, we haven't updated this library to take advantage of all the newest TCA features. We'd like to wait until we move the dependencies library to its own repo so that this package doesn't need to depend on TCA at all.

In the meantime you can create your own dependency key to register a location client with TCA.

**Reply by knight6700** [reply]
@mbrandonw i applied this solution and it's working fine with 

```
extension LocationManager: TestDependencyKey {
    public static let previewValue: Self = .live
    public static let testValue: Self = .failing
}

extension LocationManager: DependencyKey {
    public static let liveValue: Self = .live
}

extension DependencyValues {
    var locationManager: LocationManager {
        get { self[LocationManager.self] }
        set { self[LocationManager.self] = newValue }
    }
}
```
`    @Dependency(\.locationManager) var locationManager: LocationManager
`

**Reply by mbrandonw** [reply]
Great to hear! And hopefully soon we will get this library updated to use Swift's async tools instead of Combine. 😁

**Reply by knight6700** [reply]
@mbrandonw  awesome
